darts/tests/test_timeseries.py

```diff
@@ -603,40 +604,64 @@ def check_intersect(other, start_, end_, freq_):
             s_int_idx = series.slice_intersect_times(other, copy=False)
             assert s_int.time_index.equals(s_int_idx)
 
+        def check_intersect_sequence(series, other, start_, end_, freq_):
+            intersected_series = intersect([series, other])
+            s_int = intersected_series[0]
+            o_int = intersected_series[1]
+
+            assert intersected_series == [
+                series.slice_intersect(other),
+                other.slice_intersect(series),
+            ]
+
+            if start_ is None:  # empty slice
+                assert len(s_int) == 0
+                assert len(o_int) == 0
+                return
+
+            assert s_int.start_time() == o_int.start_time() == start_
+            assert s_int.end_time() == o_int.end_time() == end_
+            assert s_int.freq == o_int.freq == freq_
```

Comment:
This here is redundant, since we know already that `intersect([...])` `series.slice_intersect(...)` are identical (from above). Since you call this check always after the check for `TimeSeries.slice_intersect()`, we can assume that everything is okay